submariner/centralized-control-plane.md

```diff
@@ -0,0 +1,298 @@
+# Centralized Service Control Plane
+
+## Summary
+
+Current Submariner service management follows the model and guidelines
+ defined in [MCS](https://github.com/kubernetes-sigs/mcs-api). Specifically the
+ current design has these attributes:
+
+- Exported services are automatically reflected into all clusters in the ClusterSet;
+- Service are exported and imported using the same namespace and name; and
+- Service exports are managed from workload clusters.
+
+The above works well in an environment where clusters are used by a single
+ administrative domain and services are commonly shared across all clusters.
+ For example, when a company runs clusters as the runtime infrastructure and developers can
+ deploy to any cluster for availability, redundancy, or geographic proximity. However, it is
+ less suited in a deployment model where clusters are owned and operated by different teams,
+ each having independent control and configuration (see next section for elaboration).
+ For those deployments, we would like to propose a different approach for Service management,
+ that allows:
+
+1. Independent service naming (i.e., allow use of different names in different clusters).
+1. Selective imports (i.e., import services into a subset of clusters in the ClusterSet).
+1. Centralized control over exporting and importing (i.e., defined in Broker, not
+  workload clusters).
+
+The design proposal attempts to achieve the above with minimal changes to workload
+ clusters, especially with respect to data plane configuration. Ideally, the control
+ plane changes proposed will be confined to the Broker cluster only.
+
+## Additional Background and Context
+
+Based on work we've done in Istio (see for example [this](https://istio.io/latest/blog/2019/isolated-clusters/)),
+ we identified two common usage models for clusters:
+
+- **Shared Infrastructure**: clusters are (mostly) identically configured (e.g., namespaces,
+  RBAC, etc.). Each team is allocated the same set of namespaces across all clusters and can
+  deploy to any cluster. The boundary between teams is the namespace. This is the model assumed
+  by MCS and Submariner.
+- **Federated**: clusters are used as the boundary between teams (e.g., they may even be
+  billed to different departments). Each team can only deploy to a subset of the clusters
```

Comment:
Is it expected to have full L3 connectivity (e.g. each pod can reach each other pod/service across the cluster-set) in such a scenario?

(I'm asking about the networking aspect on ly in terms of routing, assume that no ACLs exist to block the traffic)

Reply:
without ACLs the **exposed** service's pods can be reached. Same as Submariner - you can access a service that has an export/import and is not blocked.

Reply:
Then on the L3 connectivity aspect there aren't any expected changes IIUC, and all the discussion is regarding how service discovery is handled.

It's just that the way this section is described is not clear about this point, and seems like the expectation is isolation even on the L3 level.